presto-spi/src/main/java/com/facebook/presto/spi/statistics/Reliability.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.spi.statistics;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class Reliability
+{
+    private final boolean fullyReliable;
```

Comment:
double values `0.0` and `1.0` should have exact representation in double. So maybe instead of having this field you could `reliabilityLevel` set to `1.0`. Having another variable causes some issues, for example when someone creates `Reliability` with value `1.0` then this flag will be set to `false` while stats are fully reliable. 

Also don't you need to have `fullUnreliable` field as well?